src/components/addNewBook.js

```diff
@@ -1,11 +1,46 @@
+import { useDispatch } from 'react-redux';
+import { useState } from 'react';
+import { addBook } from '../redux/books/booksSlice';
+
 export default function AddNewBook() {
+  const [values, setValue] = useState({});
+  const dispatch = useDispatch();
+
+  function handleChange(e) {
+    const { name } = e.target;
+    const { value } = e.target;
+    setValue((values) => ({
+      ...values, item_id: crypto.randomUUID(), [name]: value,
+    }));
+  }
+
+  function handleSubmit(e) {
+    e.preventDefault();
+    dispatch(addBook(values));
+    setValue({});
+  }
```

Comment:
**OPTIONAL** - You have done well so far, but to improve the UX of your project, I suggest you make it impossible to add books when any of the input fields are empty, please fix this 🙏